homeassistant/components/klyqa/config_flow.py

```diff
@@ -0,0 +1,307 @@
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if self.get_klyqa():
+            # already logged in from platform or other way
+            return self.async_abort(reason="single_instance_allowed")
+
+        await self.init()
+
+        login_failed = False
+
+        if user_input is None or login_failed:
+            return await self._show_setup_form()
+
+        self.username = str(user_input[CONF_USERNAME])
+        self.password = str(user_input[CONF_PASSWORD])
+        self.scan_interval = int(user_input[CONF_SCAN_INTERVAL])
+        self.sync_rooms = bool(user_input[CONF_SYNC_ROOMS])
+        self.polling = bool(user_input[CONF_POLLING])
+        self.host = str(user_input[CONF_HOST])
+
+        return await self._async_klyqa_login(step_id="user")
+
+    async def _async_klyqa_login(self, step_id: str) -> FlowResult:
+        """Handle login with Klyqa."""
+        errors = {}
+
+        try:
+
+            klyqa: HAKlyqaAccount = HAKlyqaAccount(
+                None,
+                None,
+                self.username,
+                self.password,
+                self.host,
+                self.hass,
+                sync_rooms=self.sync_rooms,
+                polling=self.polling,
+                scan_interval=self.scan_interval,
+            )
+
+            login = self.hass.async_run_job(
+                klyqa.login,
+            )
+            if login:
+                await asyncio.wait_for(login, timeout=30)
+            else:
+                raise Exception()
+            self.klyqa = klyqa
+
+        except (ConnectTimeout, HTTPError) as exception:
+            LOGGER.error("Unable to connect to Klyqa: %s", exception)
+            errors = {"base": "cannot_connect"}
+
+        except Exception as exception:  # pylint: disable=bare-except,broad-except
+            LOGGER.error("Unable to connect to Klyqa: %s", exception)
+            errors = {"base": "cannot_connect"}
+
+        if not self.klyqa or not self.klyqa.access_token:
+            errors = {"base": "cannot_connect"}
+
+        if errors:
+            return await self._show_setup_form(errors)
+
+        return await self._async_create_entry()
+
+    async def _async_create_entry(self) -> FlowResult:
+        """Create the config entry."""
+        config_data = {
+            CONF_USERNAME: self.username,
+            CONF_PASSWORD: self.password,
+            CONF_SCAN_INTERVAL: self.scan_interval,
+            CONF_SYNC_ROOMS: self.sync_rooms,
+            CONF_POLLING: self.polling,
+            CONF_HOST: self.host,
+        }
+        existing_entry = await self.async_set_unique_id(self.username)
+
+        if existing_entry:
+            self.hass.config_entries.async_update_entry(
+                existing_entry, data=config_data
+            )
+            # Reload the Klyqa config entry otherwise devices will remain unavailable
```

Comment:
This should not be needed here. Instead, let the integration listen for config entry changes and reload when it changes instead.